Source/WebCore/platform/graphics/cocoa/WebCoreCALayerExtras.mm

```diff
@@ -79,13 +79,14 @@ - (void)_web_setLayerTopLeftPosition:(CGPoint)position
     [self setPosition:newPosition];
 }
 
-+ (CALayer *)_web_renderLayerWithContextID:(uint32_t)contextID
++ (CALayer *)_web_renderLayerWithContextID:(uint32_t)contextID shouldPreserveFlip:(bool)preservesFlip
 {
     CALayerHost *layerHost = [CALayerHost layer];
 #ifndef NDEBUG
     [layerHost setName:@"Hosting layer"];
 #endif
     layerHost.contextId = contextID;
+    layerHost.preservesFlip = preservesFlip;
```

Comment:
is it harmful to make this always true? (I can't tell if it's "preserve geometryFlipped, which might only be set in the PDFPlugin case" or "this will always cause a flip")

Reply:
This is only set true for PlatformCALayerRemoteCustom. And setting this value doesn't actually flip the geometry, it just preserves the value that's set across layer hosting boundaries.